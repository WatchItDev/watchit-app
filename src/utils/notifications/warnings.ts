/**
 * Enum of our known warning text.
 */
export enum WARNING {
  NO_WALLET_AUTHORIZATION = "NO_WALLET_AUTHORIZATION",
  NO_WALLET_CONNECTED = "NO_WALLET_CONNECTED",
  INVALID_DEPOSIT_AMOUNT = "INVALID_DEPOSIT_AMOUNT",
  INVALID_WITHDRAW_AMOUNT = "INVALID_WITHDRAW_AMOUNT",
  INVALID_WALLET_ADDRESS = "INVALID_WALLET_ADDRESS",
  BUNDLER_UNAVAILABLE = "BUNDLER_UNAVAILABLE",
}

/**
 * Error messages dictionary mapping our ERROR_NAMES to user-facing strings.
 */
export const WARNING_MESSAGES: Record<WARNING, string> = {
  [WARNING.NO_WALLET_AUTHORIZATION]: "No wallet authorization received.",
  [WARNING.NO_WALLET_CONNECTED]: "No wallet connected.",
  [WARNING.INVALID_DEPOSIT_AMOUNT]: "Invalid deposit amount.",
  [WARNING.INVALID_WITHDRAW_AMOUNT]: "Invalid withdraw amount.",
  [WARNING.INVALID_WALLET_ADDRESS]: "Invalid wallet address.",
  [WARNING.BUNDLER_UNAVAILABLE]: "Session expired. Please login again.",
};
