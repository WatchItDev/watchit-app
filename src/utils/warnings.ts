/**
 * Enum of our known error names.
 */
export enum WARNING {
  NO_WALLET_AUTHORIZATION = 'NO_WALLET_AUTHORIZATION',
  NO_WALLET_CONNECTED = 'NO_WALLET_CONNECTED',
  INVALID_DEPOSIT_AMOUNT = 'INVALID_DEPOSIT_AMOUNT',
}

/**
 * Error messages dictionary mapping our ERROR_NAMES to user-facing strings.
 */
export const WARNING_MESSAGES: Record<WARNING, string> = {
  [WARNING.NO_WALLET_AUTHORIZATION]: 'No wallet authorization received.',
  [WARNING.NO_WALLET_CONNECTED]: 'No wallet connected.',
  [WARNING.INVALID_DEPOSIT_AMOUNT]: 'Invalid deposit amount.',
};
