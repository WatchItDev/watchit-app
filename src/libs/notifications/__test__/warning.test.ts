import { WARNING, WARNING_MESSAGES } from '../warnings';

describe('[UTILS]: Warnings', () => {
  describe('WARNING_MESSAGES', () => {
    const containsPlaceholders = (message: string): boolean =>
      message.includes('{') && message.includes('}');

    it('should match the snapshot of all warning messages', () => {
      expect(WARNING_MESSAGES).toMatchSnapshot();
    });

    it('should have a message for every warning type', () => {
      Object.values(WARNING).forEach((warningKey) => {
        expect(WARNING_MESSAGES[warningKey as WARNING]).toBeDefined();
        expect(typeof WARNING_MESSAGES[warningKey as WARNING]).toBe('string');
        expect(WARNING_MESSAGES[warningKey as WARNING].length).toBeGreaterThan(
          0,
        );
      });
    });

    it('should not have extra messages that do not correspond to warning types', () => {
      Object.keys(WARNING_MESSAGES).forEach((messageKey) => {
        expect(Object.values(WARNING)).toContain(messageKey);
      });
    });

    it('should have the correct message for NO_WALLET_AUTHORIZATION warning', () => {
      expect(WARNING_MESSAGES[WARNING.NO_WALLET_AUTHORIZATION]).toBe(
        'No wallet authorization received.',
      );
    });

    it('should have the correct message for BUNDLER_UNAVAILABLE warning', () => {
      expect(WARNING_MESSAGES[WARNING.BUNDLER_UNAVAILABLE]).toBe(
        'Session expired. Please login again.',
      );
    });

    it('should not contain placeholders in any warning messages', () => {
      Object.values(WARNING_MESSAGES).forEach((message) => {
        expect(containsPlaceholders(message)).toBe(false);
      });
    });

    it('should have consistent wallet-related warning messages', () => {
      const walletWarnings = [
        WARNING.NO_WALLET_AUTHORIZATION,
        WARNING.NO_WALLET_CONNECTED,
        WARNING.INVALID_WALLET_ADDRESS,
      ];

      walletWarnings.forEach((warning) => {
        expect(WARNING_MESSAGES[warning]).toBeDefined();
        expect(WARNING_MESSAGES[warning].toLowerCase()).toMatch(
          /wallet|address/,
        );
      });
    });

    it('should have consistent amount-related warning messages', () => {
      const amountWarnings = [
        WARNING.INVALID_DEPOSIT_AMOUNT,
        WARNING.INVALID_WITHDRAW_AMOUNT,
      ];

      amountWarnings.forEach((warning) => {
        expect(WARNING_MESSAGES[warning]).toBeDefined();
        expect(WARNING_MESSAGES[warning].toLowerCase()).toMatch(/amount/);
      });
    });
  });
});
