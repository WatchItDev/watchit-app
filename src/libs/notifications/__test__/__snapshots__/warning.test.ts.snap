// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[UTILS]: Warnings > WARNING_MESSAGES > should match the snapshot of all warning messages 1`] = `
{
  "BUNDLER_UNAVAILABLE": "Session expired. Please login again.",
  "INVALID_DEPOSIT_AMOUNT": "Invalid deposit amount.",
  "INVALID_WALLET_ADDRESS": "Invalid wallet address.",
  "INVALID_WITHDRAW_AMOUNT": "Invalid withdraw amount.",
  "NO_WALLET_AUTHORIZATION": "No wallet authorization received.",
  "NO_WALLET_CONNECTED": "No wallet connected.",
}
`;
